server/src/test/java/org/elasticsearch/common/StringsTests.java

```diff
@@ -111,4 +115,52 @@ public void testSplitStringToSet() {
         assertEquals(Strings.tokenizeByCommaToSet("   aa   "), Sets.newHashSet("aa"));
         assertEquals(Strings.tokenizeByCommaToSet("   "), Sets.newHashSet());
     }
+
+    public void testCollectionToDelimitedStringWithLimit() {
+        final String delimiter = randomFrom("", ",", ", ", "/");
+        final String prefix = randomFrom("", "[");
+        final String suffix = randomFrom("", "]");
+
+        final int count = between(0, 100);
+        final List<String> strings = new ArrayList<>(count);
+        while (strings.size() < count) {
+            // avoid starting with a sequence of empty appends, it makes the assertions much messier
+            final int minLength = strings.isEmpty() && delimiter.isEmpty() && prefix.isEmpty() && suffix.isEmpty() ? 1 : 0;
+            strings.add(randomAlphaOfLength(between(minLength, 10)));
+        }
+
+        final StringBuilder stringBuilder = new StringBuilder();
+        Strings.collectionToDelimitedStringWithLimit(strings, delimiter, prefix, suffix, Integer.MAX_VALUE, stringBuilder);
+        final String fullDescription = stringBuilder.toString();
+        for (String string : strings) {
+            assertThat(fullDescription, containsString(prefix + string + suffix));
+        }
+        assertThat(fullDescription, equalTo(Strings.collectionToDelimitedString(strings, delimiter, prefix, suffix)));
+
+        stringBuilder.setLength(0);
+        Strings.collectionToDelimitedStringWithLimit(strings, delimiter, prefix, suffix, fullDescription.length(), stringBuilder);
+        assertThat(stringBuilder.toString(), equalTo(fullDescription));
+
+        stringBuilder.setLength(0);
+        Strings.collectionToDelimitedStringWithLimit(strings, delimiter, prefix, suffix, 0, stringBuilder);
+        final String completelyTruncatedDescription = stringBuilder.toString();
+
+        if (count == 0) {
```

Comment:
What do you think splitting this test? Currently it has 3 conditional assertions which makes the test a bit complicated. I think we could make the test more readable and predictable by separating it for 3 different inputs (empty request, normal request, huge request)

Reply:
Yes ok, I pushed ff09780.